src/dataCorrectionsTool/incentiveEligibilityTool.js

```diff
@@ -0,0 +1,416 @@
+                <div class="modal-header">
+                    <h5 class="modal-title" id="confirmModalHeader">Confirm Update Incentive Eligibility</h5>
+                    <button type="button" class="close" data-dismiss="modal" aria-label="Close">
+                    <span aria-hidden="true">&times;</span>
+                    </button>
+                </div>
+                <div class="modal-body">
+                    
+                </div>
+                <div class="modal-footer">
+                    <button type="button" class="btn btn-danger" data-dismiss="modal">Cancel</button>
+                    <button type="button" class="btn btn-primary" id="confirmUpdateEligibility" data-dismiss="modal">Confirm</button>
+                </div>
+                </div>
+            </div>
+        </div>
+        `;
+};
+
+
+/**
+ * Handles the payment round selection dropdown conetent
+*/
+const handlePaymentRoundSelect = (participant) => {
+    const paymentRoundElement = document.getElementById('dropdownPaymentMenu');
+    const dropdownPaymentOptions = paymentRoundElement.children;
+    const incentiveStatusText = document.getElementById('incentiveStatusText');
+    const isIncentiveEligbleNote = document.getElementById('isIncentiveEligbleNote');
+    const selectButton = document.querySelector('.selectButton');
+    const dateOfEligibilityInput = document.getElementById('dateOfEligibility');
+
+    if (!paymentRoundElement || 
+        !dropdownPaymentOptions || 
+        !incentiveStatusText || 
+        !isIncentiveEligbleNote || 
+        !selectButton || 
+        !dateOfEligibilityInput) return;
+
+    const { paymentRound, baselinePayment, eligiblePayment, yes, no } = fieldMapping; 
+
+    for (let option of dropdownPaymentOptions) {
+        option.addEventListener('click', async (e) => {
+            participantPaymentRound = e.target.dataset.payment;
+            if (participantPaymentRound === fieldMapping['baseline'].toString()) {
+                selectButton.textContent = e.target.textContent;
+
+                try {
+                    showAnimation();
+                    const isParticipantEligibleIncentiveResponse = await checkParticipantForEligibleIncentive(participant, participantPaymentRound);
+                    hideAnimation();
+
+                    const { isEligibleForIncentive, participantData} = isParticipantEligibleIncentiveResponse.data; 
+                    const currentParticipantData = isParticipantEligibleIncentiveResponse.data;
+                    localStorage.setItem('participant', JSON.stringify(currentParticipantData.participantData));
+                    
+                    // check if isParticipantEligibleIncentive.isEligibleForIncentive is true AND current participant data has "130371375.266600170.731498909" value to "NO"
+                    isEligibleForIncentiveUpdate = isEligibleForIncentive && (participantData[paymentRound][baselinePayment][eligiblePayment] === no);
```

Comment:
Is there any chance that participantData[paymentRound][baselinePayment] will have a null object along the way and needs existence checks?